### ISSUE TITLE
[RFE] truncate journals to actual utilized size on rotate

### ISSUE THREAD

**Issue by vcaputo** [systemd/systemd#17613]
With journal-file allocating space in 8MiB increments, there tends to often be a few MiB of unused space on the tail when a journal gets rotated.

On my system here for instance, all 27 of my user journals are 8MiB in size but none of them have a tail object offset much higher than 2MiB.  It adds up to > 150MiB of wasted space.

It seems relatively straightforward for the rotate process to determine the next available offset and truncate the journal to that size to reclaim this otherwise lost space.

**Comment by poettering** [comment]
makes a ton of sense

**Comment by mrc0mmand** [comment]
Reopening due to https://github.com/systemd/systemd/pull/19934 since this is definitely a nice-to-have feature.

**Comment by vcaputo** [comment]
Once this particular issue gets sorted and truncating lands, it's probably worth exploring punching holes in the unused regions of large EntryArrayObjects on archive as well.

I slapped together a little subcommand for analyzing the EntryArrayObject utilization in jio last night: `jio report entry-arrays`: https://git.pengaru.com/cgit/jio/.git/commit/?id=aafd020086ed36ba24a76ac32b35e031bc08cf82

In case anyone would like to get some numbers from some large production journals to inform how much space would likely be reclaimed by punching those holes, the new subcommand might prove handy (plus it could use testing).

**Comment by vcaputo** [comment]
@DaanDeMeyer implemented this awhile back if I'm not mistaken, closing.
